### DISCUSSION TITLE
Searching for help

### DISCUSSION THREAD

**Discussion by 6Opaul9** [lysep-corp/ButOSX#13] [Q&A] [answered]
I am sorry to annoy you again but everything in the Third party folder is red.
I don't know why

**Comment by Lyceion** [comment]
Clone the repostory. Not download it. Open terminal, go to the folder you want to download source code, and write

`git clone https://github.com/Lyceion/ButOSX.git`

to the terminal.

**Reply by 6Opaul9** [reply]
I have tried this before, but it doesn't work.
I have SDL and SDL images installed

**Reply by ghost** [reply]
You have to also clone the sub modules. To do this run ```git clone --recurse-submodules  https://github.com/Lyceion/ButOSX.git``` to read more view this [Git Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules)

**Reply by 6Opaul9** [reply]
Now I have problems with rapidjson.
I don't know how to put the rapidjson folder to system or project's include path.

**Reply by Lyceion** [reply]
Right click to discord-rpc's folder on project's solution (i mean on xcode) click add files and the window which opened, click copy files if needed and add select rapidjson's folder. Or you can just remove everything about discord on project. It's an easier solution because it's not supported atm because its pipes broken on latest macs. (For me at least)

**Reply by Lyceion** [reply]
> You have to also clone the sub modules. To do this run ```git clone --recurse-submodules  https://github.com/Lyceion/ButOSX.git``` to read more view this [Git Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) 

Yeah you are right. Normally git should do it automaticly like it does in updated latest version. But probably it's about default version of mac's git. (I'm not sure btw)

**Reply by Lyceion** [reply]
> Now I have problems with rapidjson.
> I don't know how to put the rapidjson folder to system or project's include path.
Hey @6Opaul9 !

#14

Removed discord-rpc from project. You can clone again and freely compile yourself.

**Comment by Lyceion** [comment, marked as answer]
@jakepriddle's answer:
> You have to also clone the sub modules. To do this run ```git clone --recurse-submodules  https://github.com/Lyceion/ButOSX.git``` to read more view this [Git Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) 

Also for discord, add rapdijson to the discord-rpc's folder. Or easier solution is just remove discord-rpc from the project because it's broken.

Last thing you can do is, just download the compiled version of the program on Releases section of github. 

Note: I'm not responsible anything about cheat. This project is made with educational purposes. Code is very messy. Thx!
